File: src/transformers/features/misc/MiscCreateIsPruneTransformer.ts
import { MiscIsPruneProgrammer } from "../../../programmers/misc/MiscIsPruneProgrammer";

import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace MiscCreateIsPruneTransformer {
    export const transform = GenericTransformer.factory("misc.createIsPrune")(
        (project) => (modulo) => MiscIsPruneProgrammer.write(project)(modulo),
    );
}

File: packages/transform/src/features/misc/MiscCreateIsPruneTransformer.go
package misc

import (
	coremisc "github.com/samchon/typia/packages/core/src/programmers/misc"
	internal "github.com/samchon/typia/packages/transform/src/internal"
)

var MiscCreateIsPruneTransformer = miscCreateIsPruneTransformerNamespace{}

type miscCreateIsPruneTransformerNamespace struct{}

func (miscCreateIsPruneTransformerNamespace) Transform(call internal.ITypiaCallExpression) string {
	prune := coremisc.MiscPruneProgrammer.Write(call.TypeText)
	return "((input) => { " + prune + "(input); return true; })"
}
